src/renderer/components/swap/Swap.tsx

```diff
@@ -264,37 +274,105 @@ export const Swap = ({
     [balance, changeAmount, setChangeAmountFromPercentValue]
   )
 
-  const pending = useMemo(
+  const pendingState = useMemo(
     () =>
       pipe(
-        tx,
+        txWithState,
         RD.fold(
           () => null,
           () => <Spin />,
           (e) => (
-            <Modal closable visible title={'error'} onOk={onSwapConfirmed} okText={'Retry'} onCancel={resetTx}>
-              {e.message}
-            </Modal>
-          ),
-          (r) => (
             <Modal
               closable
               visible
-              title={'success'}
-              okText={'Open transaction page'}
-              onOk={() => goToTransaction && goToTransaction(r.hash)}
+              title={intl.formatMessage({ id: 'common.error' })}
+              onOk={onSwapConfirmed}
+              okText={intl.formatMessage({ id: 'common.retry' })}
               onCancel={resetTx}>
-              {r.hash}
+              {e.message}
             </Modal>
-          )
+          ),
+          (r) =>
+            pipe(
+              r.state,
+              RD.map(
```

Comment:
[question] Why do we need to wrap everything in this pipe into a RD? Maybe I miss something, but I can't see "remote" data nor any need to have 4 states (initial, loading, error, success)  here. Isn't `Either` not enough to handle success and error states here.

Reply:
hmmmmm... makes sense

Reply:
but wait... `Either` will tell us about a transaction RESULT (error or success) but just for example with which value would you start a `wsTransfer$` stream ? when we dont know about tx status yet ?

Reply:
As discussed in Slack, we 've decided to simplify it ^ by using `Option` (instead of `RD`)